cf_units/cf_units.py

```diff
@@ -830,7 +833,47 @@ class Unit(util._OrderedHashable):
     This class also supports time and calendar defintion and manipulation.
 
     """
-    # Declare the attribute names relevant to the _OrderedHashable behaviour.
+    def _init_from_tuple(self, values):
+        for name, value in zip(self._names, values):
+            object.__setattr__(self, name, value)
+
+    def _as_tuple(self):
+        return tuple(getattr(self, name) for name in self._names)
+
+    # Provide hash semantics
+
+    def _identity(self):
+        return self._as_tuple()
+
+    def __hash__(self):
+        return hash(self._identity())
+
+    def __eq__(self, other):
+        return (isinstance(other, type(self)) and
+                self._identity() == other._identity())
+
+    def __ne__(self, other):
+        # Since we've defined __eq__ we should also define __ne__.
+        return not self == other
+
+    # Provide default ordering semantics
+
+    def __lt__(self, other):
+        a, b = self._identity(), other._identity()
+        return (a > b) - (a < b)
```

Comment:
This looks like an implementation for `__cmp__`, not `__lt__`. It can return either 1, -1, or 0.